crates/electrum/src/electrum_ext.rs

```diff
@@ -165,6 +191,19 @@ impl<E: ElectrumApi> ElectrumExt for E {
             request.outpoints,
         )?;
 
+        // Fetch previous `TxOut`s for fee calculation if flag is enabled.
+        if fetch_prev_txouts {
+            let _ = full_scan_res.graph_update.apply_update(fetch_prev_txout(
+                self,
+                &mut tx_cache,
+                full_scan_res
+                    .graph_update
+                    .full_txs()
+                    .map(|tx_node| tx_node.tx)
+                    .collect::<Vec<_>>(),
+            )?);
+        }
```

Comment:
We don't want to `fetch_prev_txout` twice (the first time is done in `full_scan` which we call here). Fetching twice is wasteful as we don't need to calculate fee of parents of relevant transactions.

Reply:
Yes, bool should be false initially for the full_scan call and we fetch prev txouts at the end.